src/upload/form/project_assets.rs

```diff
@@ -53,6 +56,10 @@ impl ServiceWorkerAssets {
             let binding = kv.binding();
             bindings.push(binding);
         }
+        for do_ns in &self.used_durable_object_namespaces {
+            let binding = do_ns.binding().expect("namespace id to be provided");
```

Comment:
I quite dislike that there is a result to be handled here :/. I think fixing this would be annoying though

Reply:
attempting to use wrangler dev currently triggers this expect. The problem is, the new pre-upload tasks I added (for initializing self-referential namespaces, and resolving DO namespaces that are specified by name) are not run for wrangler dev... it's possible to fix without a major refactor, but it's definitely an ugly wart :(

Reply:
this is fixed now